android/sdl_android/src/androidTest/java/com/smartdevicelink/managers/screen/SoftButtonManagerTests.java

```diff
@@ -402,4 +404,32 @@ public void testSoftButtonStateEquals() {
         softButtonState2 = new SoftButtonState("object1-state1", "o1s1", artwork1);
         assertEquals(softButtonState1, softButtonState2);
     }
+
+    /**
+     * Test constructing SoftButtonObject with an empty state list
+     */
+    @Test
+    public void testConstructSoftButtonObjectWithEmptyStateList() {
+        List<SoftButtonState> stateList = new ArrayList<>();
+        // Uncommenting the following lines should make the test fail
+        // SoftButtonState softButtonState = new SoftButtonState("object1-state1", "o1s1", null);
+        // stateList.add(softButtonState);
+        SoftButtonObject softButtonObject = new SoftButtonObject("hi", stateList, "Hi", null);
+
+        assertTrue(softButtonObject.getAttemptedToAssignEmptyStateList());
+    }
+
+    /**
+     * Test assigning an empty state list to existing SoftButtonObject
+     */
+    @Test
+    public void testAssignEmptyStateListToSoftButtonObject() {
+        List<SoftButtonState> stateList = new ArrayList<>();
+        SoftButtonState softButtonState = new SoftButtonState("object1-state1", "o1s1", null);
+
+        SoftButtonObject softButtonObject = new SoftButtonObject("hi", softButtonState, null);
+        // Commenting the following line should make the test fail
+        softButtonObject.setStates(stateList);
```

Comment:
setStates should be returning early if it is passed an empty list, this means the states on the object will not be updated. to test this we can make an assertion before setting the state that the list is what is expected from the constructor.
`assertEquals(Collections.singletonList(softButtonState), softButtonObject.getStates());`
and then after trying to set the states to an empty list we can make the same assertion as setStates should return early and not update the field
`assertEquals(Collections.singletonList(softButtonState), softButtonObject.getStates());`